### PR TITLE
Add more admin tests

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for next major.

Part of #1465

### CODE REVIEW

File: src/Admin/Model/UserAdmin.php [hunk 1]
```diff
@@ -20,13 +20,30 @@
...
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 
 /**
  * @phpstan-extends AbstractAdmin<\Sonata\UserBundle\Model\UserInterface>
  */
 class UserAdmin extends AbstractAdmin
 {
+    private ?UserManagerInterface $userManager;
+
+    public function setUserManager(UserManagerInterface $userManager): void
```
Comment (VincentLanglet) [inline src/Admin/Model/UserAdmin.php]: Wouldn't it better to use constructor ?

File: src/Admin/Model/UserAdmin.php [hunk 2]
```diff
@@ -20,13 +20,30 @@
...
 use Sonata\UserBundle\Form\Type\SecurityRolesType;
+use Sonata\UserBundle\Model\UserManagerInterface;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 
 /**
  * @phpstan-extends AbstractAdmin<\Sonata\UserBundle\Model\UserInterface>
  */
 class UserAdmin extends AbstractAdmin
 {
+    private ?UserManagerInterface $userManager;
```
Comment (VincentLanglet) [inline src/Admin/Model/UserAdmin.php]: Shouldn't we provide a getter or make the property protected if the class is not final ?

File: src/Admin/Model/UserAdmin.php (hunk 1 above)
Comment (jordisala1991) [reply src/Admin/Model/UserAdmin.php]: I set the min version to 4.8 of SonataAdmin and use the constructor.

File: src/Admin/Model/UserAdmin.php (hunk 2 above)
Comment (jordisala1991) [reply src/Admin/Model/UserAdmin.php]: I made it protected.

File: src/Admin/Model/UserAdmin.php [hunk 3]
```diff
@@ -20,13 +20,26 @@
...
 /**
  * @phpstan-extends AbstractAdmin<\Sonata\UserBundle\Model\UserInterface>
  */
 class UserAdmin extends AbstractAdmin
 {
+    protected UserManagerInterface $userManager;
+
+    public function __construct(UserManagerInterface $userManager)
+    {
+        $this->userManager = $userManager;
```
Comment (VincentLanglet) [inline src/Admin/Model/UserAdmin.php:36]:
```suggestion
        parent::__construct();
        
        $this->userManager = $userManager;
```

to please phpstan

Comment (jordisala1991) [reply src/Admin/Model/UserAdmin.php:36]: Added, but not sure if it is better to add to the baseline.

File: src/Admin/Model/UserAdmin.php [hunk 4]
```diff
@@ -20,13 +20,28 @@
...
  * @phpstan-extends AbstractAdmin<\Sonata\UserBundle\Model\UserInterface>
  */
 class UserAdmin extends AbstractAdmin
 {
+    protected UserManagerInterface $userManager;
+
+    public function __construct(UserManagerInterface $userManager)
+    {
+        parent::__construct();
+
```
Comment (jordisala1991) [inline src/Admin/Model/UserAdmin.php]:
```suggestion

```

File: src/Admin/Model/UserAdmin.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Admin/Model/UserAdmin.php:36]: I think it's ok to have this "useless" call so far. It will be removed for the next sonata admin major.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**